moviepy/video/io/ffmpeg_reader.py

```diff
@@ -261,7 +261,10 @@ def ffmpeg_parse_infos(filename, print_infos=False, check_duration=True):
     if check_duration:
         try:
             keyword = ('frame=' if is_GIF else 'Duration: ')
-            line = [l for l in lines if keyword in l][0]
+            if is_GIF:
+                line = [l for l in lines if keyword in l][-1]
+            else:
+                line = [l for l in lines if keyword in l][0]
```

Comment:
Instead of writing out the `line =` twice, we could probably replace L264-L267 with something like:

```python
index = -1 if is_GIF else 0
line = [l for l in lines if keyword in l][index]
```
Granted, there is a lot we could refactor in there to get more DRY, but you get the point!

Thanks for the PR, btw!